File: 1636-sort-array-by-increasing-frequency/1636-sort-array-by-increasing-frequency.java
class Solution {
    public int[] frequencySort(int[] nums) {
        int n=nums.length;
        HashMap<Integer,Integer>mp = new HashMap<>();
        for(int i=0;i<n;i++){
            mp.put(nums[i],mp.getOrDefault(nums[i],0)+1);
        
        }
        ArrayList<Map.Entry<Integer,Integer>>result=new ArrayList<>(mp.entrySet());
        result.sort((a,b)->{
            if(a.getValue()==b.getValue()){
                return b.getKey()-a.getKey();
            }
            return a.getValue()-b.getValue();
        });
        ArrayList<Integer>list = new ArrayList<>();
        for(Map.Entry<Integer,Integer>entry: result){
                for(int i=0;i<entry.getValue();i++){
                    list.add(entry.getKey());
                }
        } 
        int [] ans = new int[list.size()];

        for(int i=0;i<list.size();i++){
            ans[i]=list.get(i);
        }
        return ans;
    }
}

File: 1636-sort-array-by-increasing-frequency/1636-sort-array-by-increasing-frequency.cpp
class Solution {
public:
    vector<int> frequencySort(vector<int>& nums) {
        int n=nums.size();
        unordered_map<int,int>mp;
        for(int i=0;i<n;i++){
            mp[nums[i]]++;
        }
        vector<pair<int,int>>list(mp.begin(),mp.end());

        sort(list.begin(),list.end(),[](auto &a , auto &b){
            if(a.second==b.second){
                return b.first < a.first;
            }
            return a.second < b.second;
        });

        vector<int>ans;
        for(auto &it: list){
            for(int i=0;i<it.second;i++){
                ans.push_back(it.first);
            }
        }
        return ans;
    }
};